x-pack/plugins/maps/common/descriptor_types/style_property_descriptor_types.d.ts

```diff
@@ -182,7 +182,11 @@ export type VectorStylePropertiesDescriptor = {
   [VECTOR_STYLES.LABEL_BORDER_SIZE]?: LabelBorderSizeStylePropertyDescriptor;
 };
 
-export type VectorStyleDescriptor = {
+export type StyleDescriptor = {
+  type: string;
+};
+
+export type VectorStyleDescriptor = StyleDescriptor & {
```

Comment:
Does VectorStyleDescriptor need to extend StyleDescriptor since it just overrides the only property on StyleDescriptor?

Reply:
It makes pulling up some common logic into `AbstractStyle` easier.

Reply:
Or just make `joins` optional?